### PR TITLE
[733] Tweak ordering when specific circumstances

### PR DESCRIPTION
**Pull request by asmega**
### Context

- https://trello.com/c/j9hfb6pP/733-build-school-order-page-for-specific-circumstances

### Changes proposed in this pull request

- Tweak ordering devices when specific circumstances 

image
image

### Guidance to review

- As a school user that can order under specific circumstances
- Attempt to order devices
- Should see view changes
- Update ordered devices to > 0 but less than cap
- Should see view changes

### CODE REVIEW

**Review by fofr — APPROVED** [review]

File: app/views/school/devices/can_order_for_specific_circumstances.html.erb [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+      <% if has_ordered %>
+        <div class="govuk-panel__body govuk-!-font-size-24">
+          You’ve ordered <%= @school.std_device_allocation.devices_ordered %> of
+          <%= @school.std_device_allocation.cap %> devices
+        </div>
+      <% else %>
+        <div class="govuk-panel__body govuk-!-font-size-24">
+          You cannot order your full allocation yet
+        </div>
+      <% end %>
```
Comment (benilovj) [inline app/views/school/devices/can_order_for_specific_circumstances.html.erb:32]:
```suggestion
      <div class="govuk-panel__body govuk-!-font-size-24">
        <% if has_ordered %>
          You’ve ordered <%= @school.std_device_allocation.devices_ordered %> of
          <%= @school.std_device_allocation.cap %> devices
        <% else %>
          You cannot order your full allocation yet
        <% end %>
      </div>
```

File: spec/controllers/school/devices_controller_spec.rb [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+RSpec.describe School::DevicesController do
+  let(:user) { create(:school_user, school: school) }
+
+  before do
+    sign_in_as user
+  end
+
+  describe '#order' do
+    context 'when school state can_order' do
```
Comment (benilovj) [inline spec/controllers/school/devices_controller_spec.rb:11]:
```suggestion
    context 'when school state is can_order' do
```

File: spec/controllers/school/devices_controller_spec.rb [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+
+      context 'when has devices left to order' do
+        let(:std_device_allocation) do
+          create(:school_device_allocation,
+                 :with_std_allocation,
+                 allocation: 10,
+                 cap: 10)
+        end
+
+        it 'render can_order' do
```
Comment (benilovj) [inline spec/controllers/school/devices_controller_spec.rb:26]:
```suggestion
        it 'renders can_order' do
```

File: spec/controllers/school/devices_controller_spec.rb [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+      end
+
+      context 'when does not have devices left to order' do
+        let(:std_device_allocation) do
+          create(:school_device_allocation,
+                 :with_std_allocation,
+                 cap: 0)
+        end
+
+        it 'render can_order' do
```
Comment (benilovj) [inline spec/controllers/school/devices_controller_spec.rb:39]:
```suggestion
        it 'renders cannot_order' do
```

File: spec/controllers/school/devices_controller_spec.rb [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+          get :order
+          expect(controller).to render_template('school/devices/cannot_order')
+        end
+      end
+    end
+
+    context 'when school state can_order_for_specific_circumstances' do
```
Comment (benilovj) [inline spec/controllers/school/devices_controller_spec.rb:46]:
```suggestion
    context 'when school state is can_order_for_specific_circumstances' do
```

**Review by benilovj — APPROVED** [review]
Just some minor suggestions.

File: spec/controllers/school/devices_controller_spec.rb (hunk 5 above)
Comment (asmega) [reply spec/controllers/school/devices_controller_spec.rb]: done

File: spec/controllers/school/devices_controller_spec.rb (hunk 4 above)
Comment (asmega) [reply spec/controllers/school/devices_controller_spec.rb]: done

File: spec/controllers/school/devices_controller_spec.rb (hunk 3 above)
Comment (asmega) [reply spec/controllers/school/devices_controller_spec.rb]: done

File: spec/controllers/school/devices_controller_spec.rb (hunk 2 above)
Comment (asmega) [reply spec/controllers/school/devices_controller_spec.rb]: done

File: app/views/school/devices/can_order_for_specific_circumstances.html.erb (hunk 1 above)
Comment (asmega) [reply app/views/school/devices/can_order_for_specific_circumstances.html.erb]: done

**Pull request merged**